### PR TITLE
feat: add renderToString function

### PR DESCRIPTION
**Pull request by cexbrayat**
It is now possible to use `renderToString` to test the SSR rendering of a component.

To do so, use `renderToString(MyComponent)`.
`renderToString` returns a `Promise<string>` with the HTML rendered.

```ts
  it('returns correct html with pre-fetched data on server', async () => {
    const Component = defineComponent({
      template: '<div>{{ text }}</div>',
      setup() {
        const text = ref<string | null>(null)
        onServerPrefetch(async () => {
          text.value = await fakeFetch('onServerPrefetch')
        })
        return { text }
      }
    })

    const contents = await renderToString(Component)

    expect(contents).toBe('<div>onServerPrefetch</div>')
  })
```

### CODE REVIEW

**Comment by cexbrayat** [comment]
@lmiller1990 @freazlike This is a rebased/tidied up version of #1572 
(@wobsoriano is still marked as the author of the commit 😉)

**Comment by cexbrayat** [comment]
I forgot to mention I added types tests and a bit of documentation

**Review by freakzlike — APPROVED** [review]
Looks good and I think it's a good starting point. 👍

File: src/renderToString.ts [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+    Defaults
+  >,
+  options?: MountingOptions<
+    Partial<Defaults> & Omit<Props & PublicProps, keyof Defaults>,
+    D
+  > &
+    Record<string, any>
+): Promise<string>
+
+// component declared by vue-tsc ScriptSetup
```
Comment (lmiller1990) [inline src/renderToString.ts:125]: Did we copy paste this entire thing? Is it possible to share it somehow?

**Review by lmiller1990 — APPROVED** [review]
Let's ship it and start to get feedback! Thanks @cexbrayat and @websoriano

File: src/renderToString.ts (hunk 1 above)
Comment (cexbrayat) [reply src/renderToString.ts:125]: @lmiller1990 Yes, it's nearly the same as mount, but returns a `Promise<string>`. I tried to extract it with a mapped type, but sadly all the tricks I found did not work (you can map one signature, but the overloads aren't mapped)😞

**Pull request merged**